scipy/stats/_continuous_distns.py

```diff
@@ -6973,6 +6973,171 @@ def _stats(self, a):
+            if not data.min() > floc:
+                raise FitDataError('powerlaw', 0, 1)
+            if fscale is not None and not data.max() <= floc + fscale:
+                raise FitDataError('powerlaw', 0, 1)
+
+        if fscale is not None:
+            if fscale <= 0:
+                raise ValueError("Negative or zero `fscale` is outside the "
+                                 "range allowed by the distribution.")
+            if fscale <= data.ptp():
+                msg = "`fscale` must be greater than the range of data."
+                raise ValueError(msg)
+
+        def get_shape(data, loc, scale):
+            # The first-order necessary condition on `shape` can be solved in
+            # closed form. It can be used no matter the assumption of the
+            # value of the shape.
+            return -len(data) / np.sum(np.log((data - loc)/scale))
+
+        def get_scale(data, loc):
+            # analytical solution for `scale` based on the location.
+            # It can be used no matter the assumption of the value of the
+            # shape.
+            return data.max() - loc
+
+        # 1) The location and scale are both fixed. Analytically determine the
+        # shape.
+        if fscale is not None and floc is not None:
+            return get_shape(data, floc, fscale), floc, fscale
+
+        # 2) The scale is fixed. The scale is one of two possible analytical
+        # solutions. Choose the one with better log-likelihood.
+        if fscale is not None:
+            args = [data, (self._fitstart(data),)]
+            func = self._reduce_func(args, {})[1]
+
+            # using `data.min()` as the optimal location
+            loc_lt1 = np.nextafter(data.min(), -np.inf)
+            shape_lt1 = fshape or get_shape(data, loc_lt1, fscale)
+            ll_lt1 = func((shape_lt1, loc_lt1, fscale), data)
+
+            # using `data.max() - scale` as the optimal location
+            loc_gt1 = np.nextafter(data.max() - fscale, np.inf)
+            shape_gt1 = fshape or get_shape(data, loc_gt1, fscale)
+            ll_gt1 = func((shape_gt1, loc_gt1, fscale), data)
+
+            if ll_lt1 < ll_gt1:
+                return shape_lt1, loc_lt1, fscale
+            else:
+                return shape_gt1, loc_gt1, fscale
+
+        # 3) The location is fixed. Return the analytical scale and the
+        # analytical (or fixed) shape.
+        if floc is not None:
+            scale = get_scale(data, floc)
+            shape = fshape or get_shape(data, floc, scale)
+            return shape, floc, scale
+
+        # 4) location, scale, and shape are all free. Attempt to fit under the
+        # assumption that `shape <= 1` analytically.
+        loc = np.nextafter(data.min(), -np.inf)
+        scale = np.nextafter(get_scale(data, loc), np.inf)
+        shape = fshape or get_shape(data, loc, scale)
+
+        # If the `shape <= 1`, then fitting has succeeded with the assumption
+        # `shape <= 1`.
```

Comment:
As @chrisb83 points out [here](https://github.com/scipy/scipy/pull/13053/files#r894995909). it's possible that there are multiple local optima  We haven't found such a case with scale free (and perhaps there are none), but we did run into this problem with scale fixed (case 2), so just like we did there, we may need to fit with strategies 4 _and_ 5, then return the better fit.